calculator.py

```diff
@@ -1,45 +1,76 @@
 import sys
-
 from calculation import calculate_discount, calculate_amount_by_distance, calculate_amount_by_time
 from Units import Unit
 import errors
+from error_messages import error_messages
 
+def get_valid_input(prompt, error_key, valid_func):
+    while True:
+        user_input = input(prompt).strip()
+        if valid_func(user_input):
+            return user_input
+        print(error_messages[error_key])
 
 def main():
     while True:
-        print("Möchtest du deine Fahrt per zurückgelegter Strecke oder Zeit berechnen?")
-        choice = input().lower()
-
-        match choice:
-            case Unit.STRECKE.value:
-                distance = float(input("Zurückgelegte Strecke eingeben:\n"))
-                unit = input("Kilometer oder Meter?\n").strip().lower()
-                amount = calculate_amount_by_distance(distance, unit)
-            case Unit.ZEIT.value:
-                time = float(input("Gefahrene Dauer eingeben:\n"))
-                unit = input("Stunden oder Minuten?\n").strip().lower()
-                amount = calculate_amount_by_time(time, unit)
-            case _:
-                print("Falsche Angabe. Programm wird neugestaret.")
-                continue
+        choice = get_valid_input(
+            "Möchtest du deine Fahrt per zurückgelegter Strecke oder Zeit berechnen?\n",
+            'invalid_choice',
+            lambda x: x in [Unit.STRECKE.value, Unit.ZEIT.value]
```

Comment:
Kannst du mir Lambda erklären?

Reply:
Lambda-Funktion, die ein Argument x annimmt und True zurückgibt, wenn x in der Liste [Unit.STRECKE.value, Unit.ZEIT.value] ist und andernfalls False, wenn also nicht in der Liste [ ] ist, erhalten wir die Fehlermeldung